shell/platform/windows/platform_handler_win32.h

```diff
@@ -15,11 +15,26 @@ namespace flutter {
 
 class FlutterWindowsView;
 
+// A public interface for ScopedClipboard, so that it can be injected into
+// PlatformHandlerWin32.
+class ScopedClipboardInterface {
+ public:
+  virtual int Open(HWND window) = 0;
+
+  virtual bool HasString() = 0;
+
+  virtual std::variant<std::wstring, int> GetString() = 0;
+
+  virtual int SetString(const std::wstring string) = 0;
+};
+
 // Win32 implementation of PlatformHandler.
 class PlatformHandlerWin32 : public PlatformHandler {
  public:
-  explicit PlatformHandlerWin32(BinaryMessenger* messenger,
-                                FlutterWindowsView* view);
+  explicit PlatformHandlerWin32(
+      BinaryMessenger* messenger,
+      FlutterWindowsView* view,
+      std::optional<ScopedClipboardInterface*> clipboard_reference);
```

Comment:
@cbracken Do you have a preference as the primary owner of this code on having constructor-based DI, vs. a private setter with a friend class, for things that will only ever be overridden for test mocks?

Reply:
I'm going to go ahead and merge, but @cbracken if you would prefer it a different way when you see this then I'm happy to open a new PR to fix it.